syncd/syncd_flex_counter.cpp

```diff
@@ -143,19 +229,20 @@ void FlexCounter::removeQueue(
 
     fc.m_queueAttrIdsMap.erase(attrIter);
 
-    // Stop flex counter thread if counter IDs map is empty
+    // Remove flex counter if counter IDs map is empty
     if (fc.m_queueCounterIdsMap.empty() && fc.m_portCounterIdsMap.empty() && fc.m_queueAttrIdsMap.empty())
     {
-        fc.endFlexCounterThread();
+        removeInstance(pollInterval);
     }
 }
 
 void FlexCounter::addPortCounterPlugin(
-        _In_ std::string sha)
+        _In_ std::string sha,
+        _In_ uint32_t pollInterval)
 {
     SWSS_LOG_ENTER();
 
-    FlexCounter &fc = getInstance();
+    FlexCounter &fc = getInstance(pollInterval);
```

Comment:
>getInstance [](start = 22, length = 11)

remove instance?

Reply:
Why do we need to remove the instance after creating? 
The port/queue list will be added to the instance. 
At least, we don't know if instance is just created or it has been already created.